### PR TITLE
OZ-464: Ozone with ERPNext data initialization

### PR DESCRIPTION
**Pull request by corneliouzbett**
Issue: https://mekomsolutions.atlassian.net/browse/OZ-464

This PR adds ERPNext data initialization. The main change is the addition of the `generate-import-script.groovy` script, which is responsible for generating a shell script to import CSV files into the ERPNext instance.

The `generate-import-script.groovy` script sorts the CSV files based on the domain order, operation order, and then alphabetically within each domain. The domain order and operation order are defined within the script, and the sorting ensures that the data is imported into the ERPNext instance consistently and predictably. 

The CSV files, are located in the `${project.basedir}/configs/erpnext/initializer_config` directory, following the naming convention `<type>_<doctype>.csv`. The type can be either 'insert' or 'update', indicating the operation to be performed, and the doctype specifies the domain of the data.

### CODE REVIEW

File: distro/configs/erpnext/initializer_config/sql/erpnext-v15.12.2-init-db.sql.gz
Comment (corneliouzbett) [inline distro/configs/erpnext/initializer_config/sql/erpnext-v15.12.2-init-db.sql.gz:1]: For this iteration, we will have to maintain the `init-db.sql.gz` file. This is due to ERPNext's data initialization process, which includes the creation of default price lists, territories, countries, currencies, etc and only occurs after the completion of the setup wizard. See [here](https://github.com/frappe/erpnext/blob/develop/erpnext/setup/setup_wizard/setup_wizard.py#L81) and also [defaults_setup](https://github.com/frappe/erpnext/blob/develop/erpnext/setup/setup_wizard/operations/defaults_setup.py)

File: scripts/erpnext/data_import/generate-import-script.groovy [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    [domainIndex, operationIndex]
+}
+
+log.info("Found ${orderedCsvFiles.size()} ERPNext Configuration CSV files to import")
+
+def scriptFile = "${project.build.directory}/${project.artifactId}-${project.version}/binaries/erpnext/scripts/data-import.sh" as Object
+
+ant.echo(file: scriptFile, message: "#!/bin/bash\n", append: false)
+ant.echo(file: scriptFile, message: "\n", append: true)
```
Comment (ibacher) [inline scripts/erpnext/data_import/generate-import-script.groovy:74]: Would this be equivalent?

```suggestion
ant.echo(file: scriptFile, message: "#!/bin/bash\n\n", append: false)
```

File: scripts/erpnext/data_import/generate-import-script.groovy [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+/**
+ * For each sorted CSV file, generate a shell script command to import the file into the ERPNext instance.
+ * The command is appended to the script file.*/
+orderedCsvFiles.each { file ->
+    def filePath = file.toString()
+    def startIndex = filePath.indexOf("/initializer_config/") + "/initializer_config/".length()
+    def strippedFilePath = filePath.substring(startIndex)
+
+    // Assumption: File name format is <type>_<doctype>.csv
+    // Example: "insert_lead.csv, update_uom.csv"
```
Comment (ibacher) [inline scripts/erpnext/data_import/generate-import-script.groovy:85]: Alright, initially, I was skeptical of the `capitalize` stuff for domain / doc type, but I'm guessing this is required. My outstanding question would be: does `update_uom.csv` actually work? It seems like in the PR its `update_UOM.csv` and I'm guessing that the capitalization is a little hard to get right for that case...

**Review by ibacher — APPROVED** [review]
Couple of qs, but LGTM.

File: distro/configs/erpnext/initializer_config/price_list/insert_price_list.csv [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ID,Price List Name,Currency,Enabled,Buying,Selling,Price Not UOM Dependent
+"8264d8a8-2ca7-4edb-824b-d4af75a58f6c",Standard Discount,USD,1,0,1,0
```
Comment (ibacher) [inline distro/configs/erpnext/initializer_config/price_list/insert_price_list.csv:2]: Additional q: why are the quotes required for UUIDs but not, e.g., for CIEL-style codes like `1513AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`

File: scripts/erpnext/data_import/generate-import-script.groovy (hunk 2 above)
Comment (corneliouzbett) [reply scripts/erpnext/data_import/generate-import-script.groovy:85]: > does update_uom.csv actually work?

No, it doesn't work. The doctype for this is UOM. Yes, is a little hard to get right, so I went with the simple approach.

File: distro/configs/erpnext/initializer_config/price_list/insert_price_list.csv (hunk 3 above)
Comment (corneliouzbett) [reply distro/configs/erpnext/initializer_config/price_list/insert_price_list.csv:2]: The import template comes with quotes for the ID column to take care of cases where the ID has spaces. Quotes aren't required for column ID If the ID has no spaces. (to be confirmed.)

**Review by mks-d — APPROVED** [review]
@rbuisson to provide a post-mortem review, but let's merge this to move things forward.

**Pull request merged**